### PR TITLE
Install fix

### PR DESCRIPTION
**Pull request by aydun**
2 commits here:
- reformat code using Civi coding standards
- fix a problem on installation where the install script looks in the wrong location to find civimembershiplevels.php and civimembershiptypes.php

Shouldn't the postflight file moves also be done when $type == "update" in case those files get updated?

### CODE REVIEW

File: fields/civicontributionpages.php [hunk 1]
```diff
@@ -34,35 +33,34 @@ class JFormFieldCiviContributionPages extends JFormField
...
    */
-  protected function getInput()
-  {
+  protected function getInput() {
     // Initialize variables.
     $html = array();
 
     // Initiate CiviCRM
-    require_once JPATH_ROOT.'/'.'administrator/components/com_civicrm/civicrm.settings.php';
+    require_once JPATH_ROOT . '/' . 'administrator/components/com_civicrm/civicrm.settings.php';
```
Comment (lcdservices) [inline fields/civicontributionpages.php]: why don't you just combine the forward slash with /administrator. no need to break it out. that's an artifact from when we were using the joomla directory separator constant.

File: fields/civimembershiplevels.php [hunk 2]
```diff
@@ -15,72 +15,70 @@
...
+    // Initialize variables.
+    $html = array();
+    $attr = '';
 
-		// Initiate CiviCRM
-		require_once JPATH_ROOT.'/'.'administrator/components/com_civicrm/civicrm.settings.php';
-		require_once 'CRM/Core/Config.php';
-		$config =& CRM_Core_Config::singleton( );
+    // Initiate CiviCRM
+    require_once JPATH_ROOT . '/' . 'administrator/components/com_civicrm/civicrm.settings.php';
```
Comment (lcdservices) [inline fields/civimembershiplevels.php]: go ahead and combine the forward slash here too

File: fields/civimembershiplevels.php [hunk 3]
```diff
@@ -15,72 +15,70 @@
...
+    // Get the field options.
+    $options = array();
+    $options[] = JHTML::_('select.option', '0', JText::_('- Select Membership Level -'));
+    $query = 'SELECT id, name FROM civicrm_membership_status WHERE is_active = 1 ORDER BY id';
+    $dao = CRM_Core_DAO::executeQuery($query);
+    while ($dao->fetch()) {
+      $options[] = JHTML::_('select.option', $dao->id, $dao->name);
     }
-	
-		// Create a read-only list (no name) with a hidden input to store the value.
```
Comment (lcdservices) [inline fields/civimembershiplevels.php]: why are you removing this? I haven't fully traced the code. can you explain what you found?

File: fields/civimembershiptypes.php [hunk 4]
```diff
@@ -36,44 +35,43 @@ class JFormFieldCiviMembershipTypes extends JFormField
...
    */
-  protected function getInput()
-  {
+  protected function getInput() {
     // Initialize variables.
     $html = array();
 
     // Initiate CiviCRM
-    require_once JPATH_ROOT.'/'.'administrator/components/com_civicrm/civicrm.settings.php';
+    require_once JPATH_ROOT . '/' . 'administrator/components/com_civicrm/civicrm.settings.php';
```
Comment (lcdservices) [inline fields/civimembershiptypes.php]: combine slash

File: install.php [hunk 5]
```diff
@@ -6,99 +6,89 @@
...
+   * method to run after an install/update/uninstall method
+   *
+   * @return void
+   */
+  function postflight($type, $parent) {
+    // $parent is the class calling this method
+    // $type is the type of change (install, update or discover_install)
+    // File move is in post flight as we have to wait on the installer having installed it first.
+    //
+    if ($type == "install") {
```
Comment (lcdservices) [inline install.php]: I agree. update this to include both install and update (us in_array)

File: rules/civimembershiptypes.php [hunk 6]
```diff
@@ -13,41 +13,39 @@
...
+  public function test(&$element, $value, $group = NULL, &$input = NULL, &$form = NULL) {
     /*
      * Here we match the value with a specific format. You may also use any kind of validation,
      * If you need a value of another field as well from the same form then use the following method:
      * $userId = ($input instanceof JRegistry) ? $input->get('id') : '0';
      * this gived you the value of the Id field
      */
 
-    require_once JPATH_ROOT.'/'.'administrator/components/com_civicrm/civicrm.settings.php';
+    require_once JPATH_ROOT . '/' . 'administrator/components/com_civicrm/civicrm.settings.php';
```
Comment (lcdservices) [inline rules/civimembershiptypes.php]: combine slash

**Comment by lcdservices** [comment]
since you're doing code cleanup, I suggested a few additional changes. I also agree they should be moved during install and update.

re: code style --
once could argue, since this is a Joomla plugin, it should follow the Joomla code pattern -- which it "sort of" does. periodically I diff the main authentication function against core Joomla and typically adhere to that code styling. modifictions/custom functions tend to be in civicrm code style, as that's what I work in the most.

anyway, I've fine with the code style changes. but some might revert for the main function when I next diff it against the core joomla plugin.

File: fields/civimembershiplevels.php (hunk 2 above)
Comment (aydun) [reply fields/civimembershiplevels.php]: ok

File: fields/civimembershiptypes.php (hunk 4 above)
Comment (aydun) [reply fields/civimembershiptypes.php]: ok

File: rules/civimembershiptypes.php (hunk 6 above)
Comment (aydun) [reply rules/civimembershiptypes.php]: ok

Comment (aydun) [reply rules/civimembershiptypes.php]: ok

File: install.php (hunk 5 above)
Comment (aydun) [reply install.php]: ok

**Comment by aydun** [comment]
Ok, I've pushed those various changes.
No problem if they revert - or if you prefer the Joomla pattern.  The main reason for changing anything was to fix $path1 in postflight() and atom painted the screen red with formatting fixes!

**Pull request merged**